### PR TITLE
fix: use ancestor inputs first in RBF withdrawals

### PR DESCRIPTION
**Pull request by bradleystachurski**
Fixes https://github.com/fedimint/fedimint/issues/5453

The implications of this bug are severe, so please review with caution. Fixing the implementation for RBF withdrawals is one approach, however it's worth considering if disabling RBF withdrawals is a better approach. If we do go with this fix, it requires all peers to shutdown and upgrade at the same time, which is already planned for 0.4. This is due to a change in the coin selection algorithm which will cause a consensus failure if peers are on different versions.

## Context

For RBF transactions we need to first spend the inputs used in the original transaction, then consider additional UTXOs if the original inputs are insufficient to cover the new feerate. Our coin selection algorithm has a bug that constructs the transaction using new UTXOs for RBF transactions. This creates two separate on-chain transactions that can be mined since we are not spending any of the same inputs as the ancestor transaction. Thankfully, the audit fails preventing the user from double spending the ecash, however this will halt any federation that has multiple UTXOs where a user initiates an RBF withdrawal.

The existing RBF withdrawal test succeeds since we initiated only a single deposit, which causes the coin selection algorithm to consider the same UTXO. The tests have been upgraded to consider multiple UTXOs.

### CODE REVIEW

File: modules/fedimint-wallet-server/src/lib.rs [hunk 1]
```diff
@@ -1423,15 +1423,15 @@ impl<'a> StatelessWallet<'a> {
         // Ensure deterministic ordering of UTXOs for all peers
         included_utxos.sort_by_key(|(_, utxo)| utxo.amount);
         remaining_utxos.sort_by_key(|(_, utxo)| utxo.amount);
-        included_utxos.extend(remaining_utxos);
+        remaining_utxos.extend(included_utxos);
```
Comment (bradleystachurski) [inline modules/fedimint-wallet-server/src/lib.rs:1426]: Subtle bug :smile:

Comment (elsirion) [reply modules/fedimint-wallet-server/src/lib.rs:1426]: Uff, easy to miss indeed :(

**Comment by elsirion** [comment]
I think we agreed yesterday to remove RBF withdraw functionality, right (details tbd)? So let's not merge but it's interesting to look at to understand the bug.

**Comment by dpc** [comment]
> perhaps using meta?

Reminder: meta is only for ... metadata about the Federation for external consumption. It was not designed as an input controlling consensus itself, which would be sort of a circular dependency and requires a bit more thought than 1 evening of hacking just because I was annoyed at clumsiness of editing external meta files.

**Comment by joschisan** [comment]
> I think we agreed yesterday to remove RBF withdraw functionality, right (details tbd)? So let's not merge but it's interesting to look at to understand the bug.

@bradleystachurski and I talked about it today and he will open a pr to remove rbf. If I understand correctly this pr would then not be merged.

**Comment by bradleystachurski** [comment]
> @bradleystachurski and I talked about it today and he will open a pr to remove rbf. If I understand correctly this pr would then not be merged.

Yep :+1:

I'll close this and open a separate PR that removes RBF withdrawals once it's ready.

**Pull request closed without merging**